lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/UnhandledTaskExceptionInTaskTest.cs

```diff
@@ -0,0 +1,231 @@
+            IObserver<IFailedTask>, IObserver<ICompletedTask>, IObserver<IFailedEvaluator>
+        {
+            private const string TaskId = "1234567";
+
+            private static readonly Logger Logger = Logger.GetLogger(typeof(UnhandledTaskExceptionInTaskTestDriver));
+
+            private readonly IEvaluatorRequestor _requestor;
+
+            private bool _shouldReceiveSerializableException;
+            private int _numFailedTasksReceived;
+
+            [Inject]
+            private UnhandledTaskExceptionInTaskTestDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitTask(GetTaskConfiguration());
+            }
+
+            public void OnNext(IFailedEvaluator value)
+            {
+                throw new Exception("Didn't expect a failed Evaluator.");
+            }
+
+            public void OnNext(IFailedTask value)
+            {
+                _numFailedTasksReceived++;
+
+                if (value.Id != TaskId)
+                {
+                    throw new Exception("Received Task ID " + value.Id + " instead of the expected Task ID " + TaskId);
+                }
+
+                // since in this test exception is thrown by Threading.Tasks.Task spawned in our Task
+                // the exception is wrapped in AggregateException with "One or more errors occurred." message
+                if (value.Message == null || value.Message != "One or more errors occurred.")
+                {
+                    throw new Exception("Exception message not properly propagated. Received message " + value.Message);
+                }
+
+                if (_shouldReceiveSerializableException)
+                {
+                    if (value.AsError() == null || !(value.AsError() is AggregateException))
+                    {
+                        throw new Exception("Outer exception should have been an AggregateException. " + value.AsError());
+                    }
+                    var inner = value.AsError().InnerException;
+
+                    if (inner == null || !(inner is TestSerializableException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
+                    }
+
+                    if (inner.Message != ExpectedExceptionMessage)
+                    {
+                        throw new Exception("Incorrect Exception message, got message: " + value.AsError().Message);
+                    }
+
+                    value.GetActiveContext().Value.Dispose();
+
+                    if (_numFailedTasksReceived == 2)
```

Comment:
Put the log before context dispose to make sure it is logged before driver is shut down.